unix-socket-lib/racket/unix-socket-tcp-unit.rkt

```diff
@@ -0,0 +1,55 @@
+#lang racket/base
+
+(require net/tcp-sig
+         racket/unit
+         "unix-socket.rkt")
+
+(provide
+ make-unix-socket-tcp@)
+
+(define (make-unix-socket-tcp@ path)
+  (unit
+    (import)
+    (export tcp^)
+
+    (define (tcp-abandon-port p)
+      (error 'tcp-abandon-port "not supported"))
+
+    (define (tcp-accept l)
+      (unix-socket-accept l))
+
+    (define (tcp-accept/enable-break l)
```

Comment:
This implementation of `tcp-accept/enable-break` doesn't satisfy the guarantee that either the accept or the break occurs but not both. I think the following code should fix it (but I haven't tested it):

    (define (tcp-accept/enable-break l)
      (apply values (sync/enable-break (unix-socket-accept-evt l)))

Reply:
Thanks! I misunderstood the docs of `tcp-accept/enable-break`. I'll fix that and try to add a test.

Reply:
I made the change you suggested and I added a test that finds the problem with the old implementation under CS and BC if I add a loop like

```racket
  (time
   (let loop ([n 1000000000])
     (unless (zero? n)
       (loop (sub1 n)))))
```

to the top of `accept-poll/check` to increase the likelihood that the timing works out.

The new implementation never fails the test under BC, but it does fail it under CS. Looking through the documentation and at the implementation of `accept-evt`, I don't see any reason why this shouldn't work under CS as well so I think this could be some sort of bug in CS. I'll take another look at it tomorrow in case my test is broken and open an issue in the Racket repo if it's not.

Reply:
That was indeed a Racket CS bug that got fixed in https://github.com/racket/racket/commit/4d0aa443b135d1d6fa3111ce934f63fe352aa60e . Since that commit, I can no longer reproduce the failure. Matthew added a similar test to the Racket test suite, so I went ahead and dropped mine from this repo.